### PR TITLE
Use explicit return types in bigtable_instance_admin_snippets.cc.

### PR DESCRIPTION
**Pull request by coryan**
Part of the work for #2604.

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/2616)

### CODE REVIEW

File: google/cloud/bigtable/examples/bigtable_instance_admin_snippets.cc [hunk 1]
```diff
@@ -272,17 +284,18 @@ void CreateCluster(google::cloud::bigtable::InstanceAdmin instance_admin,
...
   namespace cbt = google::cloud::bigtable;
+  using google::cloud::future;
   using google::cloud::StatusOr;
   [](cbt::InstanceAdmin instance_admin, std::string instance_id,
      std::string cluster_id, std::string zone) {
     auto cluster_config = cbt::ClusterConfig(zone, 3, cbt::ClusterConfig::HDD);
-    auto future_cluster = instance_admin.CreateCluster(
-        cluster_config, cbt::InstanceId(instance_id),
-        cbt::ClusterId(cluster_id));
+    future<StatusOr<google::bigtable::admin::v2::Cluster>> future_cluster =
```
Comment (devbww) [inline google/cloud/bigtable/examples/bigtable_instance_admin_snippets.cc]: Nit: Other examples above were like `f_cluster`.

File: google/cloud/bigtable/examples/bigtable_instance_admin_snippets.cc [hunk 2]
```diff
@@ -643,20 +663,21 @@ void UpdateAppProfileRoutingAny(
...
+  using google::cloud::future;
+  using google::cloud::StatusOr;
   [](cbt::InstanceAdmin instance_admin, std::string instance_id,
      std::string profile_id) {
-    auto profile_future = instance_admin.UpdateAppProfile(
-        cbt::InstanceId(instance_id), cbt::AppProfileId(profile_id),
-        cbt::AppProfileUpdateConfig()
-            .set_multi_cluster_use_any()
-            .set_ignore_warnings(true));
+    future<StatusOr<google::bigtable::admin::v2::AppProfile>> profile_future =
```
Comment (devbww) [inline google/cloud/bigtable/examples/bigtable_instance_admin_snippets.cc]: Nit: `profile_future` -v `future_cluster`.

File: google/cloud/bigtable/examples/bigtable_instance_admin_snippets.cc (hunk 1 above)
Comment (coryan) [reply google/cloud/bigtable/examples/bigtable_instance_admin_snippets.cc]: Good point. I settled for `foo_future` everywhere, more explicit is better than less explicit for these examples.

File: google/cloud/bigtable/examples/bigtable_instance_admin_snippets.cc (hunk 2 above)
Comment (coryan) [reply google/cloud/bigtable/examples/bigtable_instance_admin_snippets.cc]: See above.

**Comment by coryan** [comment]
PTAL.

**Pull request merged**